src/conventions/h-group/clue-interpretation/interpret-clue.js

```diff
@@ -79,9 +79,30 @@ function resolve_clue(game, old_game, action, inf_possibilities, focused_card) {
 	const old_inferred = old_game.common.thoughts[focused_card.order].inferred;
 
 	focus_thoughts.inferred = focus_thoughts.inferred.intersect(inf_possibilities);
+	focus_thoughts.self_connection = false;
 
-	for (const { connections, suitIndex, rank, save, interp } of inf_possibilities) {
+	for (const { connections, suitIndex, rank, save, interp, self_connection } of inf_possibilities) {
 		const inference = { suitIndex, rank };
+		focus_thoughts.self_connection ||= (self_connection?.length > 0);
+
+		// A finesse is considered important if it could only have been given by this player.
+		if (connections.some(connection => connection.type == 'finesse')) {
+			for (let i = (giver + 1) % state.numPlayers; i != target; i = (i + 1) % state.numPlayers) {
+				if (i == target)
+					continue;
+
+				if (i == connections[0].reacting) {
+					// The clue must be given before the first reacting player.
+					// If we get to the first reacting player, assume important.
+					action.important = true;
+					break;
+				}
+
+				// A player can't give a finesse if they didn't know some card in the finesse.
+				if (connections.some(connection => connection.reacting == i && connection.type != 'known'))
+					continue;
```

Comment:
How is this condition necessary? It seems if control flow gets to this point, the loop is about to continue anyways. Can we just simplify this entire loop into `inBetween(state.numPlayers, connections[0].reacting, giver, target)`?

Reply:
Yeah sorry, I meant to add a last condition which breaks if we find a player who could give the finesse.